Sources/Workspace/Workspace.swift

```diff
@@ -606,9 +606,9 @@ extension Workspace {
         revision: Revision? = nil,
         checkoutBranch: String? = nil,
         observabilityScope: ObservabilityScope
-    ) {
+    ) async {
```

Comment:
This doesn't account for reentrancy which is now possible with these functions becoming `async`. What happens when `edit` is called again before previous `edit` has finished?

IMO `Workspace` should get a `state` stored property that's asserted on to have `.idle` value and set to a different value indicating that the workspace is no longer idle when  `async` call is started.